src/components/browse/components/FacetList/FacetTermsList.js

```diff
@@ -184,6 +209,16 @@ export class Term extends React.PureComponent {
                 </a>
             </li>
         );
+
+        if (searchType === 'basic') {
+            // if (!searchText) { return facetListItem; }
+            const termAlreadyFiltered = _.any(filteredTerms, function (item) { return item.key === term.key; });
```

Comment:
This seems quite imperformant -- think about how to refactor...

There might be 100+ terms on any view (most likely not all would be SAYT terms/facets, but still many might be). If each of these 100+ terms then iterates over other filtered terms (is this same size as all terms, potentially?) then we get performance of like O(n^2) and with enough Terms, UI might start to hang upon changes/as-enter-text.

Would at least have another (memoized, probably) function for this check or have `filteredTerms` be an object for faster lookups (rather than iterating over each term in filteredTerms using `_.any`).

Reply:
Agree on performance note.

In reference to that edit, I think this is a bigger question that would need discussion at a UI meeting. I completely get what you're saying - most of the lists of items we'd be using for local search are small enough that SAYT isn't super necessary, but I think there might be a couple of cases where it really would be preferable (long lists of country names, for instance). So it's sort of a balancing act; not necessary for everything, but probably better to have for others.